### DISCUSSION TITLE
Allow line breaks in vertical line at code folds

### DISCUSSION THREAD

**Discussion by advice2020** [geany/geany#3986] [Ideas]
Often when applications contain "code folds" they leave a space / break in the verical line between them as a visiual indicator that one code fold ends and another begins, which I feel can be helpful visually to see the separation.
However I noticed that in Geany the vertical line continues as one solid long line and shows a little horiztonal line indicator at the ends of each code fold.
Suggesting that maybe if willing could add a setting to be able to show a space between code folds.

Thank You

EDIT:
I have added some more information about this in my post below if anyone is interested.

**Comment by elextr** [comment]
Try the possible settings of the `Tools->Configuration Files->filetypes.common->folding_style` setting.

I suspect that the vertical line from a higher level fold combines with the end corner for a nested fold.  "Somebody" would have to find the code that sets the vertical line and (optionally) not write it on lines that are the end of a nested fold.  The code may not be in Geany it may be in the Scintilla or Lexilla projects Geany uses, this is left as an exercise for the reader.

**Comment by advice2020** [comment]
When I had noticed this occurring I was in some longer XML files, which made me think that that is just how Geany handled code folds.  I recently realized that when opening files in other languages that Geany does include the breaks in this vertical line but I think it is based on the what level hierarchy.  Only the items closest to the left with no indents get these vertical breaks, all the other indents within them get the little horizontal line I had mentioned with no break.  I actually like this, the only thing is, unfortunately often all the content within XML files a usually within one level and all the other levels are indented within it giving the appearance that Geany does not break vertical lines at code folds at all.  Unfortunately this can make XML files confusing because user does not get to see any breaks at the folders so it just comes off as one continous group of code.

Having these breaks can be very helpful to give eyes a break and know where one group of code ends and another begins. That is why I also made the following request #3986 or adding an option to show horizontal lines at code breaks as well.

I wish there was an existing way to adjust at what level these breaks would occur, but would have to be a setting based on only applying to specific language formats (for example XML) because if it was a setting that applied to ALL languages code folds then you would lose the Geany default that I mentioned I like above.

"The code may not be in Geany it may be in the Scintilla or Lexilla projects Geany uses"
I do not have the knowledge to find out where this code occurs personally and I doubt any other users will as well.

If any other users have any ideas, please let me know.
